File: app/src/main/java/com/jikexueyuan/mobile/address/StringConvertFactory.java
package com.jikexueyuan.mobile.address;

import com.squareup.okhttp.RequestBody;
import com.squareup.okhttp.ResponseBody;

import java.lang.annotation.Annotation;
import java.lang.reflect.Type;

import retrofit.Converter;

/**
 * Created by aven on 10/22/15.
 */
public class StringConvertFactory extends Converter.Factory {
    @Override
    public Converter<ResponseBody, ?> fromResponseBody(Type type, Annotation[] annotations) {
        return new StringResponseBodyConverter();
    }

    @Override
    public Converter<?, RequestBody> toRequestBody(Type type, Annotation[] annotations) {
        return super.toRequestBody(type, annotations);
    }
}

File: app/src/main/kotlin/com/jikexueyuan/mobile/address/api/StringConvertFactory.kt
package com.jikexueyuan.mobile.address.api

import com.squareup.okhttp.RequestBody
import com.squareup.okhttp.ResponseBody
import java.lang.reflect.Type

import retrofit.Converter

/**
 * Created by aven on 10/22/15.
 */
class StringConvertFactory : Converter.Factory() {
    override fun fromResponseBody(type: Type?, annotations: Array<Annotation>?): Converter<ResponseBody, *> {
        return StringResponseBodyConverter()
    }

    override fun toRequestBody(type: Type?, annotations: Array<Annotation>?): Converter<*, RequestBody> {
        return super.toRequestBody(type, annotations)
    }
}
